upgrade/util/util_test.go

```diff
@@ -46,3 +63,156 @@ func TestProgressMonitorInc(t *testing.T) {
 		t.Fatalf(`targetValue = %v, expectedTargetValue = %v`, targetValue, expectedTargetValue)
 	}
 }
+
+func TestCheckUpgradePathSupported(t *testing.T) {
+	testCases := []struct {
+		name           string
+		currentVersion string
+		upgradeVersion string
+		expectError    bool
+	}{
+		{
+			name:           "new install",
+			currentVersion: "",
+			upgradeVersion: "v1.5.0",
+			expectError:    false,
+		},
+		{
+			name:           "invalid upgrade version",
+			currentVersion: "v1.5.0",
+			upgradeVersion: "v1.5.x",
+			expectError:    true,
+		},
+		{
+			name:           "major version downgrade",
+			currentVersion: "v2.0.0",
+			upgradeVersion: "v1.5.0",
+			expectError:    true,
+		},
+		{
+			name:           "upgrade across two major versions",
+			currentVersion: "v1.5.0",
+			upgradeVersion: "v3.0.0",
+			expectError:    true,
+		},
+		{
+			name:           "major version upgrade with lower minor version",
+			currentVersion: "v1.5.0",
+			upgradeVersion: "v2.0.0",
+			expectError:    true,
+		},
+		{
+			name:           "major version upgrade with upgradable minor version",
+			currentVersion: "v1.30.0",
+			upgradeVersion: "v2.0.0",
+			expectError:    false,
+		},
+		{
+			name:           "major version upgrade from non-GA version",
+			currentVersion: "v1.31.0-rc2",
+			upgradeVersion: "v2.0.0",
+			expectError:    true,
+		},
+		{
+			name:           "major version upgrade to non-GA version",
+			currentVersion: "v1.31.0",
+			upgradeVersion: "v2.0.0-rc3",
+			expectError:    false,
+		},
+		{
+			name:           "upgrade across two minor versions",
+			currentVersion: "v1.3.0-dev",
+			upgradeVersion: "v1.5.0",
+			expectError:    true,
+		},
+		{
+			name:           "upgradable minor version from non-GA version",
+			currentVersion: "v1.4.1-rc1",
+			upgradeVersion: "v1.5.0",
+			expectError:    true,
+		},
+		{
+			name:           "upgradable minor version from non-GA version to non-GA version",
+			currentVersion: "v1.4.1-rc1",
+			upgradeVersion: "v1.5.0-rc2",
+			expectError:    true,
+		},
+		{
+			name:           "upgradable minor version to non-GA version",
+			currentVersion: "v1.4.2",
+			upgradeVersion: "v1.5.0-rc2",
+			expectError:    false,
```

Comment:
This should be error. Can't upgrade to any non-GA version.

Reply:
Removed the test case.